applicationset/services/pull_request/gitlab.go

```diff
@@ -54,7 +54,7 @@ func NewGitLabService(ctx context.Context, token, url, project string, labels []
 	}, nil
 }
 
-func (g *GitLabService) List(ctx context.Context) ([]*PullRequest, error) {
+func (g *GitLabService) List(_ context.Context) ([]*PullRequest, error) {
```

Comment:
Instead of dropping the param, we should use it on the list request: `g.client.MergeRequests.ListProjectMergeRequests(g.project, opts, gitlab.WithContext(ctx))`.